x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -177,40 +198,45 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
     }
   }, [caseDetailsRefresh]);
 
-  if (!expandedEvent?.eventId) {
-    return null;
-  }
-
-  return isFlyoutView ? (
-    <>
-      <EuiFlyoutHeader hasBorder={isHostIsolationPanelOpen}>
-        {isHostIsolationPanelOpen ? (
-          backToAlertDetailsLink
-        ) : (
+  const renderFlyoutHeader = useMemo(() => {
```

Comment:
can we extract it to the standalone component?

Reply:
I would leave this as it is, there is too many dependencies within this component. And I do not want to pass tens of props down.